apps/nfid-frontend/src/integration/swap/service/swap-service.ts

```
@@ -22,6 +25,10 @@ export class SwapService {
       map.set(icpSwapShroff.getSwapName(), icpSwapShroff)
     } catch (e) {
       map.set(SwapName.ICPSwap, undefined)
+
+      if (e instanceof ServiceUnavailableError) {
```

we can move kong logic to catch of icpswap and remove PROVIDERS_QUANTITY, serviceErrors...

---

kongswap has separate try/catch statement. And I need to throw the error in case all the providers are unavailable

---

if first throws error then we try to fetch it from second, isn’t it right flow? In case second also fails we can throw your error